lib/plugins/index.js

```diff
@@ -6,36 +6,41 @@ const {validatePlugin, validateStep, loadPlugin, parseConfig} = require('./utils
 const pipeline = require('./pipeline');
 const normalize = require('./normalize');
 
-module.exports = (context, pluginsPath) => {
+module.exports = async (context, pluginsPath) => {
   let {options, logger} = context;
   const errors = [];
 
   const plugins = options.plugins
-    ? castArray(options.plugins).reduce((plugins, plugin) => {
-        if (validatePlugin(plugin)) {
-          const [name, config] = parseConfig(plugin);
-          plugin = isString(name) ? loadPlugin(context, name, pluginsPath) : name;
+    ? await castArray(options.plugins).reduce(async (plugins, plugin) => {
+      const plugins2 = await plugins;
+      if (validatePlugin(plugin)) {
+        const [name, config] = parseConfig(plugin);
+        plugin = isString(name) ? await loadPlugin(context, name, pluginsPath) : name;
 
-          if (isPlainObject(plugin)) {
-            Object.entries(plugin).forEach(([type, func]) => {
-              if (PLUGINS_DEFINITIONS[type]) {
-                Reflect.defineProperty(func, 'pluginName', {
-                  value: isPlainObject(name) ? 'Inline plugin' : name,
-                  writable: false,
-                  enumerable: true,
-                });
-                plugins[type] = [...(plugins[type] || []), [func, config]];
-              }
-            });
-          } else {
-            errors.push(getError('EPLUGINSCONF', {plugin}));
-          }
+        console.log({plugins: options.plugins})
+        console.dir(plugin, {depth: null})
+        console.log(isPlainObject(plugin))
```

Comment:
just a reminder to remove the debug logs when the PR is ready :)

Reply:
yep 👍🏼